fix: Delete the file and recall it with Ctrl+Z, displaying an error pop-up window in dde-file-dailog

liyigang1 opened the pull request:
Failure to load the trashcore plugin resulted in no trashinfo creation failure. Use dfileinfo to read data during restore

Log: Delete the file and recall it with Ctrl+Z, displaying an error pop-up window in dde-file-dailog
Bug: https://pms.uniontech.com/bug-view-254735.html

max-lvs commented on src/plugins/common/core/dfmplugin-fileoperations/fileoperations/trashfiles/dorestoretrashfilesworker.cpp:
```diff
@@ -292,6 +286,173 @@ bool DoRestoreTrashFilesWorker::checkRestoreInfo(const QUrl &url, FileInfoPointe
     return result;
 }
 
+bool DoRestoreTrashFilesWorker::doCheckRestoreFromTrashFile(const DFileInfo &fromInfo, const FileInfoPointer &toInfo, const QString &fileName, FileInfoPointer &newTargetInfo, bool *skip)
```
代码冗余太多，需要合并优化，在filebasework里面有了。

liyigang1 committed:
fix: Delete the file and recall it with Ctrl+Z, displaying an error pop-up window in dde-file-dailog

Failure to load the trashcore plugin resulted in no trashinfo creation failure. Use dfileinfo to read data during restore

Log: Delete the file and recall it with Ctrl+Z, displaying an error pop-up window in dde-file-dailog
Bug: https://pms.uniontech.com/bug-view-254735.html

liyigang1 commented:
/merge
